packages/browser-lite/src/database.ts

```diff
@@ -0,0 +1,552 @@
+// tslint:disable:max-file-line-count
+
+import Dexie from 'dexie';
+
+export type Record = Order | MiniHeader | Metadata;
+
+export interface Options {
+    dataSourceName: string;
+    maxOrders: number;
+    maxMiniHeaders: number;
+}
+
+export interface Query<T extends Record> {
```

Comment:
I hadn't considered these "wrapper types". That naming and conversion convention was used in a very specific place in __packages/browser-lite__. It was used to wrap `core.App` functions, methods, and structs, into types that we could access from JavaScript land.

We don't really implement `JSValue` methods for anything in the `db` package right now. There are a few places where we have placeholder implementations which use `jsutil.InefficientlyConvertToJS`. I mentioned in https://github.com/0xProject/0x-mesh/pull/820 that we should look into defining `JSValue` methods as a future optimization. Are you okay with revisiting this later if we do decide to go that route? We should have some kind of test to make sure that we are converting correctly between JS and Go at that time.

Reply:
Hmm, yeah I can see how these aren't really "wrapper types" since we only use them internally. 

Yes, let's revisit this if we define `JSValue` functions. I think that the existence of a bespoke `JSValue` function is a good litmus test for whether or not conversion tests are needed. This may need to be reevaluated with very complicated Typescript typings, but it's fine with me in this case.

Reply:
> I think that the existence of a bespoke JSValue function is a good litmus test for whether or not conversion tests are needed. 

I also think we should write conversion tests if we are doing any additional conversion in JavaScript land, which in this case we are not.